dht.go

```diff
@@ -495,6 +526,54 @@ func (dht *IpfsDHT) SetMode(m DHTMode) error {
 	}
 }
 
+func (dht *IpfsDHT) dynamicModeSwitching(proc goprocess.Process) {
+	var (
+		debouncer = time.NewTimer(0)
+		target    DHTMode
+	)
+	defer debouncer.Stop()
+
+	stopTimer := func() {
+		if debouncer.Stop() {
+			return
+		}
+		select {
+		case <-debouncer.C:
+		default:
```

Comment:
Note: this is technically kind of racy because Stop doesn't wait for the timer to stop, but it shouldn't really be an issue.

Reply:
timer.Stop() is a huge headache. I never got it to work as described in godocs, so I went this route 🙃